geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometricOutputsTest.scala

```diff
@@ -0,0 +1,178 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import java.util.{Map => JMap}
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.spark.sql._
+import org.geotools.data.{DataStore, DataStoreFinder}
+import org.junit.runner.RunWith
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import scala.collection.JavaConversions._
+
+@RunWith(classOf[JUnitRunner])
+class SparkSQLGeometricOutputsTest extends Specification with LazyLogging {
+
+  "sql geometry constructors" should {
+    sequential
+
+    val dsParams: JMap[String, String] = Map("cqengine" -> "true", "geotools" -> "true")
+    var ds: DataStore = null
+    var spark: SparkSession = null
+    var sc: SQLContext = null
+
+    var df: DataFrame = null
+
+    // before
+    step {
+      ds = DataStoreFinder.getDataStore(dsParams)
+      spark = SparkSQLTestUtils.createSparkSession()
+      sc = spark.sqlContext
+
+      SparkSQLTestUtils.ingestChicago(ds)
+
+      df = spark.read
+        .format("geomesa")
+        .options(dsParams)
+        .option("geomesa.feature", "chicago")
+        .load()
+      logger.info(df.schema.treeString)
+      df.createOrReplaceTempView("chicago")
+
+      df.collect().length mustEqual 3
+    }
+
+    "st_asBinary" >> {
+      val r = sc.sql(
+        """
+          |select st_asBinary(st_geomFromWKT('POLYGON((0 0, 2 0, 2 2, 0 2, 0 0))'))
+        """.stripMargin
+      )
+      r.collect().head.getAs[Array[Byte]](0) mustEqual Array[Byte](0, 0, 0, 0, 3, 0, 0, 0, 1, 0, 0, 0, 5, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0,
+        64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0
+      )
+    }
+
+    "st_asGeoJSON" >> {
+      // st_asGeoJSON(geom) from chicago
+
+      "point" >> {
+        val r = sc.sql(
+          """
+            |select st_asGeoJSON(st_geomFromWKT('POINT(0 0)'))
+          """.stripMargin
+        )
+        r.collect().head.getAs[String](0) mustEqual
+          "{\"type\":\"Point\",\"coordinates\":[0.0,0.0]}"
```

Comment:
For quoted text like this, try out triple double quotes.  """ { "type": "Point", "coordinates":[0.0,0.0] } """

(Whitespace added to make it more readable; it'd probably cause this test to fail.)